// @framer/jurrehoutkamp.audio-player/code/AudioPlayer.tsx
import * as React from "react";
import { PropertyControls, ControlType, RenderTarget } from "framer";

interface Props {
  width: number;
  height: number;
  src: string;
  autoplay: boolean;
  loop: boolean;
  muted: boolean;
  controls: boolean;
}

export class AudioPlayer extends React.Component<Props> {
  static defaultProps = {
    width: 320,
    height: 40,
    src:
      "https://946663360620-us-east-1-static-files.s3.amazonaws.com/audio-snippets/store.mp3",
    autoplay: false,
    loop: false,
    muted: false,
    controls: true
  };

  static propertyControls: PropertyControls<Props> = {
    src: { type: ControlType.String, title: "Source" },
    autoplay: { type: ControlType.Boolean, title: "Play" },
    loop: { type: ControlType.Boolean, title: "Loop" },
    muted: { type: ControlType.Boolean, title: "Muted" },
    controls: { type: ControlType.Boolean, title: "Show" }
  };

  render() {
    let { width, height, src, autoplay, loop, muted, controls } = this.props;

    if (RenderTarget.current() != RenderTarget.preview) {
      controls = true;
      muted = false;
    }

    return (
      <div
        style={{
          width: width,
          height: height,
          overflow: "hidden",
          backgroundColor: "rgba(255, 255, 255, 0)",
          color: "rgba(255, 255, 255, 0)",
          fontSize: "14px"
        }}
      >
        <audio
          key={`#${autoplay}`}
          autoPlay={autoplay}
          loop={loop}
          muted={muted}
          controls={controls}
          height={80}
          preload="auto"
          style={{
            width: "100%",
            height: "100%",
            objectPosition: "50% 50%"
          }}
          src={src}
        />
        Your browser does not support HTML5 audio.
      </div>
    );
  }
  x;
}
