// @framer/ruucm-h.html-video/code/App.tsx
import { Override, Data, Color } from "framer"

const appState1 = Data({
  video0PlayerState: "stop",
  video1PlayerState: "stop",
  video2PlayerState: "stop",
})

export function Video0(props): Override {
  return {
    playerState: appState1.video0PlayerState,
  }
}

export function Video1(props): Override {
  return {
    playerState: appState1.video1PlayerState,
  }
}

export function Video2(props): Override {
  return {
    playerState: appState1.video2PlayerState,
  }
}

export function Page(props): Override {
  return {
    onChangePage: (current, prev) => {
      console.log("Current Page Number", current)

      if (current === 0) {
        appState1.video0PlayerState = "play"
        appState1.video1PlayerState = "pause"
        appState1.video2PlayerState = "pause"
      } else if (current === 1) {
        console.log("play!")
        appState1.video0PlayerState = "pause"
        appState1.video1PlayerState = "play"
        appState1.video2PlayerState = "pause"
      } else if (current === 2) {
        appState1.video0PlayerState = "pause"
        appState1.video1PlayerState = "pause"
        appState1.video2PlayerState = "play"
      }
    },
  }
}
