// @framer/ismael.sounds/code/kits.tsx
const fb = "./assets/sounds/facebook"
const go = "./assets/sounds/google"

type SoundPath = { [key: string]: string };

export const facebook : SoundPath = {
    button1: fb + "/Buttons_and_Navigation/Button 1.m4a",
    button2: fb + "/Buttons_and_Navigation/Button 2.m4a",
    button3: fb + "/Buttons_and_Navigation/Button 3.m4a",
    button4: fb + "/Buttons_and_Navigation/Button 4.m4a",
    button5: fb + "/Buttons_and_Navigation/Button 5.m4a",
    button6: fb + "/Buttons_and_Navigation/Button 6.m4a",
    button7: fb + "/Buttons_and_Navigation/Button 7.m4a",
    collapse: fb + "/Buttons_and_Navigation/Collapse.m4a",
    expand: fb + "/Buttons_and_Navigation/Expand.m4a",
    tab1: fb + "/Buttons_and_Navigation/Tab 1.m4a",
    tab2: fb + "/Buttons_and_Navigation/Tab 2.m4a",
    tab3: fb + "/Buttons_and_Navigation/Tab 3.m4a",
    complete1: fb + "/Complete_and_Success/Complete 1.m4a",
    complete2: fb + "/Complete_and_Success/Complete 2.m4a",
    complete3: fb + "/Complete_and_Success/Complete 3.m4a",
    success1: fb + "/Complete_and_Success/Success 1.m4a",
    success2: fb + "/Complete_and_Success/Success 2.m4a",
    success3: fb + "/Complete_and_Success/Success 3.m4a",
    cancel1: fb + "/Errors_and_Cancel/Cancel 1.m4a",
    cancel2: fb + "/Errors_and_Cancel/Cancel 2.m4a",
    error1: fb + "/Errors_and_Cancel/Error 1.m4a",
    error2: fb + "/Errors_and_Cancel/Error 2.m4a",
    error3: fb + "/Errors_and_Cancel/Error 3.m4a",
    error4: fb + "/Errors_and_Cancel/Error 4.m4a",
    error5: fb + "/Errors_and_Cancel/Error 5.m4a",
    alert1: fb + "/Notifications_and_Alerts/Alert 1.m4a",
    alert2: fb + "/Notifications_and_Alerts/Alert 2.m4a",
    alert3: fb + "/Notifications_and_Alerts/Alert 3.m4a",
    alert4: fb + "/Notifications_and_Alerts/Alert 4.m4a",
    alert5: fb + "/Notifications_and_Alerts/Alert 5.m4a",
    notification1: fb + "/Notifications_and_Alerts/Notification 1.m4a",
    notification2: fb + "/Notifications_and_Alerts/Notification 2.m4a",
    notification3: fb + "/Notifications_and_Alerts/Notification 3.m4a",
    notification4: fb + "/Notifications_and_Alerts/Notification 4.m4a",
    notification5: fb + "/Notifications_and_Alerts/Notification 5.m4a",
    notification6: fb + "/Notifications_and_Alerts/Notification 6.m4a",
    notification7: fb + "/Notifications_and_Alerts/Notification 7.m4a",
    notification8: fb + "/Notifications_and_Alerts/Notification 8.m4a",
    notification9: fb + "/Notifications_and_Alerts/Notification 9.m4a",
}

export const google = {
    heroDecorativeCelebration1: go + "/01_Hero_Sounds/hero_decorative-celebration-01.m4a",
    heroDecorativeCelebration2: go + "/01_Hero_Sounds/hero_decorative-celebration-02.m4a",
    heroDecorativeCelebration3: go + "/01_Hero_Sounds/hero_decorative-celebration-03.m4a",
    heroSimpleCelebration1: go + "/01_Hero_Sounds/hero_simple-celebration-01.m4a",
    heroSimpleCelebration2: go + "/01_Hero_Sounds/hero_simple-celebration-02.m4a",
    heroSimpleCelebration3: go + "/01_Hero_Sounds/hero_simple-celebration-03.m4a",
    alarmGentle: go + "/02_Alerts_and_Notifications/alarm_gentle.m4a",
    alarmHighIntensity: go + "/02_Alerts_and_Notifications/alert_high-intensity.m4a",
    alertSimple: go + "/02_Alerts_and_Notifications/alert_simple.m4a",
    notificationAmbient: go + "/02_Alerts_and_Notifications/notification_ambient.m4a",
    notificationDecorative1: go + "/02_Alerts_and_Notifications/notification_decorative-01.m4a",
    notificationDecorative2: go + "/02_Alerts_and_Notifications/notification_decorative-02.m4a",
    notificationHighIntensity: go + "/02_Alerts_and_Notifications/notification_high-intensity.m4a",
    notificationSimple1: go + "/02_Alerts_and_Notifications/notification_simple-01.m4a",
    notificationSimple2: go + "/02_Alerts_and_Notifications/notification_simple-02.m4a",
    ringtoneMinimal: go + "/02_Alerts_and_Notifications/ringtone_minimal.m4a",
    navigationBackwardSelectionMinimal: go + "/03_Primary_System_Sounds/navigation_backward-selection-minimal.m4a",
    navigationBackwardSelection: go + "/03_Primary_System_Sounds/navigation_backward-selection.m4a",
    navigationForwardSelectionMinimal: go + "/03_Primary_System_Sounds/navigation_forward-selection-minimal.m4a",
    navigationForwardSelection: go + "/03_Primary_System_Sounds/navigation_forward-selection.m4a",
    notificationHoverTap: go + "/03_Primary_System_Sounds/navigation_hover-tap.m4a",
    navigationSelectionCompleteCelebration: go + "/03_Primary_System_Sounds/navigation_selection-complete-celebration.m4a",
    stateChangeConfirmDown: go + "/03_Primary_System_Sounds/state-change_confirm-down.m4a",
    stateChangeConfirmUp: go + "/03_Primary_System_Sounds/state-change_confirm-up.m4a",
    uiCameraShutter: go + "/03_Primary_System_Sounds/ui_camera-shutter.m4a",
    uiLock: go + "/03_Primary_System_Sounds/ui_lock.m4a",
    uiUnlock: go + "/03_Primary_System_Sounds/ui_unlock.m4a",
    uiTapVariant1: go + "/03_Primary_System_Sounds/ui_tap-variant-01.m4a",
    uiTapVariant2: go + "/03_Primary_System_Sounds/ui_tap-variant-02.m4a",
    uiTapVariant3: go + "/03_Primary_System_Sounds/ui_tap-variant-03.m4a",
    uiTapVariant4: go + "/03_Primary_System_Sounds/ui_tap-variant-04.m4a",
    alertError1: go + "/04_Secondary_System_Sounds/alert_error-01.m4a",
    alertError2: go + "/04_Secondary_System_Sounds/alert_error-02.m4a",
    alertError3: go + "/04_Secondary_System_Sounds/alert_error-03.m4a",
    navigationCancel: go + "/04_Secondary_System_Sounds/navigation-cancel.m4a",
    navigationTransitionLeft: go + "/04_Secondary_System_Sounds/navigation_transition-left.m4a",
    navigationTransitionRight: go + "/04_Secondary_System_Sounds/navigation_transition-right.m4a",
    navigationUnavailableSelection: go + "/04_Secondary_System_Sounds/navigation_unavailable-selection.m4a",
    uiLoading: go + "/04_Secondary_System_Sounds/ui_loading.m4a",
    uiRefreshFeed: go + "/04_Secondary_System_Sounds/ui_refresh-feed.m4a",
}
