// @framer/ismael.sounds/code/Overrides.tsx
import { Data, Override } from "framer"
import { playSound, sounds } from "./Sounds"

const data = Data({
    toggle: false,
})

export function Draggable(): Override {
    return {
        drag: true,
    }
}

export function MessageIn(): Override {
    return {
        x: 200,
        opacity: 0,
        animate: data.toggle && {
            x: 0,
            opacity: 1,
        },
        transition: {
            duration: 0.1,
            type: "spring",
            stiffness: 110,
        },
        onTapS() {
            data.toggle = true
        },
        onAnimationStart() {
            playSound(sounds.collapse)
        },
    }
}

export function MainButton(): Override {
    return {
        whileTap: {
            scale: 0.9,
        },
        onTapStart() {
            playSound(sounds.uiLock)
        },
    }
}

export function MainButtonAbsolute(): Override {
    const file =
        "https://isma.uy/demo/sounds/google/02_Alerts_and_Notifications/alert_simple.wav"
    const sound = new Audio(file)

    return {
        whileTap: {
            scale: 0.9,
        },
        onTapStart() {
            playSound(sound)
        },
    }
}
